src/query/pools/query_pools.go

```diff
@@ -79,7 +79,13 @@ func BuildWorkerPools(
 		return workerPool
 	})
 
-	return objectPool, instrumentOptions
+	writePoolSize := cfg.WriteWorkerPoolSize
+	if writePoolSize == 0 {
+		writePoolSize = defaultWorkerPoolSize
+	}
+
+	writeWorkerPool := xsync.NewWorkerPool(writePoolSize)
+	return objectPool, writeWorkerPool, instrumentOptions
```

Comment:
We're getting quite a few things returns here, consider bundling it? I foresee a lot more pools being added as we work on perf :P

Reply:
Wanna use the new worker pool I wrote for the m3db ingesters? Its way more performant because it doesn't allocate goroutines all the time: https://github.com/m3db/m3x/blob/master/sync/pooled_worker_pool.go

That should help fix a lot of the `runtime.morestack/runtime.growstack/runtime.copystack` in the m3coordinator flame graphs

Reply:
I'm pretty sure its API compatible too so I think you can literally just replace the constructors and be done